pygmsh/surface_base.py

```diff
@@ -4,11 +4,13 @@
 
 class SurfaceBase(object):
     _ID = 0
+    num_edges = 0
 
-    def __init__(self, id0=None):
+    def __init__(self, num_edges=0, id0=None):
```

Comment:
I had them initially in the reverse order, but that would imply that, in order to specify a name for the surface, it would be required to set also `num_edges`, that can be undesirable.
Anyway I totally understand the backward compatibility issue here, furthermore because python will silently use a string in the num_edges field without any warning.
I'll leave the final decision to you.